src/server/game/AI/CreatureAI.h

```diff
@@ -218,4 +219,51 @@ enum Permitions
     PERMIT_BASE_SPECIAL            = 800
 };
 
+class EvadeHomeCall : public BasicEvent
+{
+public:
+    explicit EvadeHomeCall(Creature* owner) : _owner(owner) { }
+
+    bool Execute(uint64 /*time*/, uint32 /*diff*/)
+    {
+        if (_owner->IsAIEnabled && _owner->GetAI() && reinterpret_cast<const SmartAI*>(_owner->GetAI()) == nullptr)
```

Comment:
then move this method to .cpp and include smartai
reinterpret_cast doesn't do what you want here - it does NOT return nullptr when the cast is invalid, only dynamic_cast does that (and to know whether the cast is valid it needs the class definition, that is why you need the include)